### ISSUE TITLE
Failed to build Dockerfile 3_1_2-4

### ISSUE THREAD

**Issue by luisfelipe3d** [pkp/docker-ojs#53]
### Expected Behavior
Build the docker image
### Current Behavior
 ``` bash
3/3 [============================] 100%
  - Installing citation-style-language/locales (1.0.0): Cloning master from cache
  - Installing symfony/polyfill-mbstring (v1.12.0): Extracting archive
  - Installing seboettg/collection (v1.3.1): Extracting archive
  - Installing myclabs/php-enum (1.7.2): Extracting archive
  - Installing seboettg/citeproc-php (v2.1.6): Extracting archive
 0/4 [>---------------------------]   0%
 3/4 [=====================>------]  75%
 4/4 [============================] 100%
Generating autoload files
/bin/sh: npm: not found
The command '/bin/sh -c composer update -d lib/pkp --no-dev && composer install 
 -d plugins/paymethod/paypal --no-dev && composer install 
 -d plugins/generic/citationStyleLanguage --no-dev && npm install -y && npm run build' 
returned a non-zero code: 127
```
### Steps to Reproduce
1. Go to versions/3_1_2-4/alpine/apache/php/
2. Run docker build -t local/ojs:3_1_2-4 .

### Details
I suspect the problem is with the **templates/dockerFiles/Dockerfile-alpine-apache-php7.template** file, the ENV PACKAGES does not contain npm package.
``` dockerfile
ENV COMPOSER_ALLOW_SUPERUSER=1  \
    SERVERNAME="localhost"      \
    HTTPS="on"                  \
    OJS_VERSION=%%OJS_VERSION%% \
    OJS_CLI_INSTALL="0"         \
    OJS_DB_HOST="localhost"     \
    OJS_DB_USER="ojs"           \
    OJS_DB_PASSWORD="ojs"       \
    OJS_DB_NAME="ojs"           \
    OJS_WEB_CONF="/etc/apache2/conf.d/ojs.conf"   \
    OJS_CONF="/var/www/html/config.inc.php"       \
    PACKAGES="supervisor dcron ttf-freefont apache2 apache2-ssl apache2-utils patch php7 php7-fpm php7-cli php7-apache2   \
             php7-zlib php7-json php7-mbstring php7-tokenizer php7-simplexml php7-phar php7-openssl    \
             php7-curl php7-mcrypt php7-pdo_mysql php7-mysqli php7-session php7-ctype php7-gd php7-xml \
             php7-dom php7-iconv curl nodejs git" \
    EXCLUDE_URL="https://raw.githubusercontent.com/pkp/docker-ojs/master/excludeVar.list"
```
I believe that this part of the file contains a typo in **npm install -y**
```dockerfile
# Install NPM and Composer Deps
RUN composer update -d lib/pkp --no-dev && \
    composer install -d plugins/paymethod/paypal --no-dev && \
    composer install -d plugins/generic/citationStyleLanguage --no-dev && \
    npm install -y && npm run build
```

### Context (Environment)
``` bash
$ lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux bookworm/sid
Release:        testing
Codename:       bookworm

$ uname -a
Linux alohomora 5.10.0-8-amd64 #1 SMP Debian 5.10.46-4 (2021-08-03) x86_64 GNU/Linux

$ docker -v
Docker version 20.10.5+dfsg1, build 55c4c88

$ docker-compose --version
docker-compose version 1.27.4, build unknown

```
Luis Leal.

**Comment by marcbria** [comment]
Thanks for the detailed report and the bug digging.
Please, attach a PR and I will merge and rebuild the images.

**Comment by marcbria** [comment]
Hi Felipe,

I checked this one and I got an error but not the same you got.

To keep the container smaller, "npm" is installed as follows: 

First we declare BUILDER apps:
https://github.com/pkp/docker-ojs/blob/7951a11892512f9a587bf8aa77d287f2b34e96e4/templates/dockerFiles/Dockerfile-alpine-apache-php73.template#L75

And after this, builder tools are installed here:
https://github.com/pkp/docker-ojs/blob/7951a11892512f9a587bf8aa77d287f2b34e96e4/templates/dockerFiles/Dockerfile-alpine-apache-php73.template#L86

Finally we remove the BUILDER tools, as far are not required in production any more.
It will make easier to create a dev-Dockerfile in future, where those tools need to be kept (or it all will all change if we finally move to multi-stage).

Opening a new issue to fix the new error at: #138 

... even I don't know if make sense to keep old images as far as I don't know if somebody is really using them.

**Comment by marcbria** [comment]
Me olvidaba. Si prefieres que hablemos en español, por mi encantado.

Un saludo,
m.

**Comment by luisfelipe3d** [comment]
> ... even I don't know if make sense to keep old images as far as I don't know if somebody is really using them.   

Maybe it's time to move this versions to legacy folder.

> It will make easier to create a dev-Dockerfile in future, where those tools need to be kept (or it all will all change if we finally move to multi-stage).

How about separating the production and development environment? It will be better on new issue to discuss about this.

**Comment by marcbria** [comment]
> Maybe it's time to move this versions to legacy folder.

Some months ago PKP show statistics about the versions people was using and we found there are still a lot of people with in old and unsuported ojs 2.x.

Once we got a stable and solid images (I'm keeping this images on my spare time), I have in mind building an "ojs universal upgrade box". Most of the work is almost done.

The idea is, you deliver your mysqldump, your public-private files and let the script do the magic to return a dockerized ojs on last version (or a list of upgrade errors).

This is why I keep those old images... we need them as an starting point to let people upgrade.

Probably I can move it to a different folder (let's call it "legacy") but I can't remove it and I will need to fix them someday.

> How about separating the production and development environment? It will be better on new issue to discuss about this.

Perfect. Let's talk about this in a new issue. 

> Hace algunos años que yo no hablo o escribo en español jajaja

Let me know if you like to practice. ;-)
